### ISSUE TITLE
Compilation fails with GCC 5.1.0

### ISSUE THREAD

**Issue by jbiel** [davisp/jiffy#89]
Compilation of this project with GCC 5.1.0 fails. We've reverted to an older version (4.9.2) for now. Thanks.

```
==> jiffy (compile)
...
Compiling c_src/double-conversion/bignum.cc
c_src/double-conversion/bignum.cc: In member function ‘void double_conversion::Bignum::AssignDecimalString(double_conversion::Vector<const char>)’:
c_src/double-conversion/bignum.cc:101:6: error: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Werror=strict-overflow]
 void Bignum::AssignDecimalString(Vector<const char> value) {
      ^
cc1plus: all warnings being treated as errors
ERROR: c++ -c  -g -Wall -Werror -O3 -g -Wall -fPIC  -I/usr/lib/erlang/lib/erl_interface-3.7.20/include -I/usr/lib/erlang/erts-6.4/include   c_src/double-conversion/bignum.cc -o c_src/double-conversion/bignum.o failed with error: 1 and output:
c_src/double-conversion/bignum.cc: In member function ‘void double_conversion::Bignum::AssignDecimalString(double_conversion::Vector<const char>)’:
c_src/double-conversion/bignum.cc:101:6: error: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Werror=strict-overflow]
 void Bignum::AssignDecimalString(Vector<const char> value) {
      ^
cc1plus: all warnings being treated as errors
```

```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-unknown-linux-gnu/5.1.0/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: /build/gcc-multilib/src/gcc-5.1.0/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --with-multilib-list=m32,m64,mx32 --enable-checking=release --enable-libmpx --with-default-libstdcxx-abi=c++98
Thread model: posix
gcc version 5.1.0 (GCC)
```

**Comment by kolargol** [comment]
same issue here...

**Comment by jbiel** [comment]
This issue can be worked around with this [commit](https://github.com/altrg/jiffy/commit/3175fee6f5113f7065dcbaf1797050ce66b9f64e) (but I'm not sure if this is the "right" thing to do.)

**Comment by kolargol** [comment]
yes that fixes compiling, and is mentioned on error, but not fixes source of the problem :)

**Comment by unbalancedparentheses** [comment]
same issue here...

**Comment by jbiel** [comment]
This is fixed in master as of 446e284; closing.
